### PR TITLE
Created Domain Model Classes and Added Hibernate Annotations

### PR DESCRIPTION
**Pull request by antoninguerre**

### CODE REVIEW

File: lib/src/main/java/ECSE428Project/model/Account.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        accountMatches = new ArrayList<>();
+    }
+
+    public Account(String aId, String aName, String aEmail, boolean verified, boolean loggedIn, int aScore, int aLevel) {
+        id = aId;
+        name = aName;
+        email = aEmail;
+        isVerified = verified;
+        isLoggedIn = loggedIn;
+        score = aScore;
```
Comment (JulietteDebray) [inline lib/src/main/java/ECSE428Project/model/Account.java:63]: what's the difference between score and level?

File: lib/src/main/java/ECSE428Project/model/Match.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+import javax.persistence.Id;
+import javax.persistence.ManyToOne;
+import javax.persistence.OneToMany;
+import javax.persistence.Table;
+import java.util.ArrayList;
+import java.util.List;
+
+@Entity
+@Table (name = "table")
+public class Match {
```
Comment (JulietteDebray) [inline lib/src/main/java/ECSE428Project/model/Match.java:17]: edit this to be round instead?

File: lib/src/main/java/ECSE428Project/model/Match.java [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    private Options options;
+
+
+    //------------------------
+    // ASSOCIATIONS
+    //------------------------
+
+    @ManyToOne(optional = false)
+    private Account account;
+
```
Comment (JulietteDebray) [inline lib/src/main/java/ECSE428Project/model/Match.java:38]: i'm still confused about player/account association to the match

File: lib/src/main/java/ECSE428Project/model/Match.java [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+        winnerId = null;
+        winnerName = null;
+        gameMode = null;
+        options = null;
+        account = null;
+        player = null;
+        rounds = new ArrayList<>();
+    }
+
+    public Match(String aMatchId, String aWinnerId, String aWinnerName, GameMode aGameMode, Options someOptions, Account anAccount, Player aPlayer) {
```
Comment (JulietteDebray) [inline lib/src/main/java/ECSE428Project/model/Match.java:62]: also don't we need a list of players?

Comment (antoninguerre) [reply lib/src/main/java/ECSE428Project/model/Match.java:62]: Yeah, that would make sense but then we need to change the UML model

File: lib/src/main/java/ECSE428Project/model/Round.java [hunk 5]
```diff
@@ -0,0 +1,70 @@
+package ECSE428Project.model;
```
Comment (JulietteDebray) [inline lib/src/main/java/ECSE428Project/model/Round.java:1]: there's round and match?

Comment (antoninguerre) [reply lib/src/main/java/ECSE428Project/model/Round.java:1]: Yes a match is composed of multiple rounds

Comment (antoninguerre) [reply lib/src/main/java/ECSE428Project/model/Round.java:1]: I just followed what was in the UML Class diagram we might want to change the terms later on but I don't think it matters too much

**Review by JulietteDebray — APPROVED** [review]

**Pull request merged**